protocol-units/execution/maptos/opt-executor/src/executor.rs

```diff
@@ -529,27 +556,30 @@ mod tests {
 		let tx_factory = TransactionFactory::new(executor.aptos_config.chain_id.clone());
 
 		// Loop to simulate the execution of multiple blocks.
-		for _ in 0..10 {
+		for i in 0..10 {
+
+			let (epoch, round) = executor.get_next_epoch_and_round().await?;
+
 			// Generate a random block ID.
 			let block_id = HashValue::random();
 			// Clone the signer from the executor for signing the metadata.
-			// let signer = executor.signer.clone();
+			let signer = executor.signer.clone();
 			// Get the current time in microseconds for the block timestamp.
-			// let current_time_micros = chrono::Utc::now().timestamp_micros() as u64;
+			let current_time_micros = chrono::Utc::now().timestamp_micros() as u64;
 
 			// Create a block metadata transaction.
-			/*let block_metadata = Transaction::BlockMetadata(BlockMetadata::new(
+			let block_metadata = Transaction::BlockMetadata(BlockMetadata::new(
 				block_id,
-				1,
-				0,
+				epoch,
+				round,
 				signer.author(),
-				vec![0],
+				vec![],
 				vec![],
 				current_time_micros,
-			));*/
+			));
 
 			// Create a state checkpoint transaction using the block ID.
-			// let state_checkpoint_tx = Transaction::StateCheckpoint(block_id.clone());
+			let state_checkpoint_tx = Transaction::StateCheckpoint(block_id.clone());
```

Comment:
This transaction is not used anywhere, was it meant to be added to any of the blocks?